### DISCUSSION TITLE
Tracking Namespaces

### DISCUSSION THREAD

**Discussion by ThePerfectComputer** [florianschanda/miss_hit#279] [Q&A]
I know that with miss_hit, you can do ``mh.register_file()``. Does this track namespaces at all? For example, a file that sits in `+dir1/+dir2/+dir3/func.m` could be referenced as `dir1.dir2.dir3.func`. Does miss hit have any concept of this?

**Comment by florianschanda** [comment]
There is some notion of this, but of course without semantic resolution this is largely irrelevant.

MH does check that you don't have things like +a/foo/bar.m because that cannot be seen, but that's pretty much it.

https://github.com/florianschanda/miss_hit/blob/f01fbf9cd43c52fdb8fe6dce162c9772637ec199/miss_hit_core/m_ast.py#L394-L396
